### ISSUE TITLE
New export type: all records with OCLC numbers

### ISSUE THREAD

**Issue by escowles** [pulibrary/bibdata#578]
We would like to create a new export for all records with OCLC numbers (e.g., everything we think we can share), stripping out the 583 field (includes private notes for RBSC items)

**Comment by hackmastera** [comment]
(Can be any sort of endpoint)

**Comment by escowles** [comment]
Notes on policy (in ITIMS team drive): https://docs.google.com/document/d/1rdPQ4G9rpG63KKoCQOnG0CdEkbP5Lo4sbIEhh9csT20/edit

**Comment by escowles** [comment]
Ivy Plus data lake proposal: https://docs.google.com/document/d/1uaMFlVA4_c4CFvGZNDZ_dlMLCCzTPmgAEQXY31YEarc/edit

**Comment by escowles** [comment]
Latest draft proposal for how this would work: https://docs.google.com/document/d/1uz6G4qvD_wnT-bhOBbOgJWhxppUT9fnRYDpdfwDsf0g/edit#heading=h.jem4s4a2v25p

TL;DR: we would have a client that would POST to a web service, with either MARC21/MARCXML in the body, or with the URL where it could be downloaded.

**Comment by hectorcorrea** [comment]
Here is a quick and dirty script that we could use to produce the files for an initial dump: 

https://gist.github.com/hectorcorrea/e1e40a312ce93c1ac5d2ec0be5502373

A more sophisticated approach would probably be desired for continuous updates to the data.

**Comment by escowles** [comment]
@hectorcorrea I was playing around with this a couple of weeks ago, writing some scripts to fetch a full dump from bibdata and do the same filtering: https://github.com/escowles/pod_scripts/

I tried in both Ruby using ruby-marc and in Java using MARC4J, to see what the relative performance was. I would be very interested to see how that stacked up to marcli, too.

**Comment by hectorcorrea** [comment]
I finally ran and compared Esmé's implementations. 

The Ruby version is way too slow, it takes about ~7 minutes per file, that translates to 13+ hours for all 100 files. 

The Java implementation is incredibly fast, it takes about 10 seconds per file. 

The Go implementation is somewhere in between, it takes 35 seconds per file.

I forked Esmé's repo and put it under https://github.com/pulibrary/pod_scripts and standardized the scripts to run each of the implementation so that it's easier to compare and contrast.

**Comment by escowles** [comment]
From @kevinreiss: There are some items with OCLC numbers that we don't loan outside of Princeton. We should be able to identify these by their shelf location codes.

**Comment by hectorcorrea** [comment]
@escowles good to know. I've just pushed a full load to POD with data from the 113 dump files: https://pod.stanford.edu/organizations/princeton. 

I used your Java program with a few modifications (https://github.com/pulibrary/pod_scripts) but we can easily tweak it to exclude certain locations and re-post the files. The files that I uploaded are old anyway (Jan/2021) and a new set would be a good idea.
